### ISSUE TITLE
Cross-Origin Resource Sharing issue

### ISSUE THREAD

**Issue by TeoTN** [amercier/promise-window#11]
The library fails to close a window from other origin when a message is posted.
Error (Chrome 55.0.2883.87)
```
Uncaught DOMException: Blocked a frame with origin "http://localhost:3000" from accessing a cross-origin frame.
    at PromiseWindow.close (http://localhost:3000/static/js/bundle.js:65614:29)
    at PromiseWindow.onPostMessage (http://localhost:3000/static/js/bundle.js:65382:13)
    at PromiseWindow._onPostMessage (http://localhost:3000/static/js/bundle.js:65549:34)
```
In order to reproduce a problem, you have to open a window with a different origin and post a message to the opener.

It could be solved in one of following ways:
* by wrapping `this.close()` in try-catch and letting user to handle different origin window closing 
* by sending a `close` message from parent to child so that child may handle closing by itself
* by allowing user to specify a boolean parameter which would prevent the library from closing the window and will call a callback function instead.

**Comment by theevs** [comment]
this._window.onclose = null; throw exception

**Comment by amercier** [comment]
Thanks for reporting this @TeoTN, I'll try to have a look at it today (CET).

**Comment by amercier** [comment]
The `this._window.onclose = null` statement is triggering the Error, not the `this._window.close()` one. Also, it doesn't seem to be required. I'm investigating further to see if removing the line entirely is safe. In this case we might not need any special treatment for cross-origin windows.

**Comment by amercier** [comment]
Fix has been released in v1.2.0 (see #12). Please reopen if the problem persists.
PS thanks @qwerty6906 too for pointing out the root cause ;)
